### DISCUSSION TITLE
Is it possible to get data from only some specific columns?

### DISCUSSION THREAD

**Discussion by devloper16** [pocketbase/pocketbase#4585] [Q&A] [answered]
I've a collection containing 13 columns I am using getFullList(filter: `user = "${user}"`) to get data from it. Is it possible to only get data from columns named user, link, type, time? possible to reduce payload size.

**Comment by ganigeorgiev** [comment, marked as answer]
You can list the fields you want to return in the `fields` parameter:

```
fields: "id,user,link,type,time"
```
(_you can find more details in the Admin UI API preview or in the general [List/Search API docs](https://pocketbase.io/docs/api-records/#listsearch-records)_).

**Reply by SusuGostoso** [reply]
Taking advantage of the topic, is it possible to create a field filter to update a user? For example, the "common" user (**role = 1**) will not be able to modify the "**role**" field. Is it possible to do that?

**Reply by ganigeorgiev** [reply]
@SusuGostoso I'm not entirely sure that I understand what do you mean. `fields` in the above is a client-side request parameter and it is used to specify a subset of the fields that should be returned in the response. If you want to apply some constraint on what collection fields should be updated you'll have to either move the hidden/private fields to a new collection OR define an API rule(s) to restrict what fields are being submitted, ex. `@request.data.role:isset = false` (see also https://pocketbase.io/docs/api-rules-and-filters/#isset-modifier). If you have further questions please open a new discussion with more details about your use case and what you are trying to do.

**Reply by devloper16** [reply]
@SusuGostoso  In my current app I am using role based user access. In users collection I've added a role column and I've constructed api rules based on their role they can perform certain actions. (ie. list all records in collection or edit a record)

You could use rule `@request.auth.role = "USER WHO CAN EDIT THE RECORD"`. Or if you want only the owner of the record to be able to edit it you can do `@request.auth.role = user.id`  but for this to work you will need a user assigned to each record (using relational records).
